src/main/java/seedu/address/logic/commands/task/EditTaskCommand.java

```diff
@@ -0,0 +1,210 @@
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Optional;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.commons.Address;
+import seedu.address.model.commons.Name;
+import seedu.address.model.task.Task;
+
+public class EditTaskCommand extends Command {
+
+    public static final String COMMAND_WORD = "edit";
+
+    public static final String MESSAGE_USAGE = Task.TYPE + " " + COMMAND_WORD
+        + ": Edits the details of the task identified "
+        + "by the index number used in the displayed contact list. "
+        + "Existing values will be overwritten by the input values.\n"
+        + "Parameters: INDEX (must be a positive integer) "
+        + "[" + PREFIX_NAME + "NAME] "
+        + "[" + PREFIX_DATE + "DATE] "
+        + "[" + PREFIX_ADDRESS + "ADDRESS] "
+        + "[" + PREFIX_WEIGHTAGE + "WEIGHTAGE] "
+        + "[" + PREFIX_REMARK + "REMARK]...\n"
+        + "Example: " + COMMAND_WORD + " 1 "
+        + PREFIX_WEIGHTAGE + "25 "
+        + PREFIX_REMARK + "New remark";
+
+    public static final String MESSAGE_EDIT_TASK_SUCCESS = "Edited Task: %1$s";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_DUPLICATE_TASK = "This task already exists in TrackIt@NUS.";
+
+    private final Index index;
+    private final EditTaskCommand.EditTaskDescriptor editTaskDescriptor;
+
+    /**
+     * @param index              of the task in the filtered contact list to edit
+     * @param editTaskDescriptor details to edit the task with
+     */
+    public EditTaskCommand(Index index, EditTaskCommand.EditTaskDescriptor editTaskDescriptor) {
+        requireNonNull(index);
+        requireNonNull(editTaskDescriptor);
+
+        this.index = index;
+        this.editTaskDescriptor = new EditTaskCommand.EditTaskDescriptor(editTaskDescriptor);
+    }
+
+    private static Task createEditedTask(Task taskToEdit, EditTaskDescriptor editTaskDescriptor) {
+        assert taskToEdit != null;
+
+        Name updatedName = editTaskDescriptor.getName().orElse(taskToEdit.getName());
+        LocalDate updatedDate = editTaskDescriptor.getDate().orElse(taskToEdit.getDate());
+        Address updatedLocation = editTaskDescriptor.getLocation().orElse(taskToEdit.getLocation());
+        double updatedWeightage = editTaskDescriptor.getWeightage().orElse(taskToEdit.getWeightage());
+        String updatedRemarks = editTaskDescriptor.getRemark().orElse(taskToEdit.getRemark());
+
+        return new Task(updatedName, updatedDate, updatedLocation, updatedWeightage, updatedRemarks);
```

Comment:
should we merge Location into Remark now?

Reply:
I think that will require some refactoring. I suggest I create an issue for it and we settle in v1.3